cli/cycpp.py

```diff
@@ -1305,8 +1352,34 @@ def impl(self, ind="  "):
             if 'derived_init' in info:
                 impl += ind + info['derived_init'] + '\n'
             else:
+                labels = info.get('alias', None)
+                if labels is None:
+                    labels = member if isinstance(t, STRING_TYPES) else [member]
                 reader = self.readers.get(t, self.readers.get(t[0], None))
-                impl += reader(member, alias, t, d, uitype, ind)
+
+                # generate condition to choose default vs given val if default exists
+                if d is not None:
+                    name = labels if isinstance(t, STRING_TYPES) else labels[0]
+                    impl += ind + 'if (sub->NMatches("{0}") > 0) {{\n'.format(name)
```

Comment:
I understand the viewpoint.  But the people 'debugging' codegen'd code are most likely cyclus core developers - the same people maintaining the code generator.  There is a non-trivial cost in code-generator readability, ease of maintainability, etc. No free lunch.  I think the cost is higher than the benefit in this case.  But as I said - fixed - just wanted to  make my opinion known :-)